hearth/lib/hearth.rb

```diff
@@ -38,6 +38,7 @@
 require_relative 'hearth/query/param_list'
 require_relative 'hearth/refreshing_identity_resolver'
 require_relative 'hearth/retry'
+require_relative 'hearth/rules_engine'
```

Comment:
Maybe this should actually be called endpoint_rules? It's not clear what rules_engine refers to (we know it's endpoints, though)

Reply:
Yeah - this was one of my open questions on the PR description.

I started with `RulesEngine` to match the naming used by Smithy for all of these.  But I agree - it is all endpoints so could be an Endpoint module instead.

Reply:
Yeah. I vote EndpointRules or even Endpoints (preferred?) as a module name. Maybe evaluate what that would look like globally and see if it makes sense.

Reply:
1+ on renaming to Endpoint-something. As someone somewhat new to the codebase, I would not have known that this file is all for endpoints

Reply:
See: https://github.com/smithy-lang/smithy-ruby/pull/170/commits/e7342a768cdf0176374ce7d2019067ac8fe0c874 for what the rename looks like.

I'm fine with using `Endpoints` as the module name since its descriptive of what its actually used for - I was mostly initially trying to match Smithy's namespace.